test/extended/templates/helpers.go

```diff
@@ -58,3 +78,90 @@ func setUser(cli *exutil.CLI, user *userapi.User) {
 		cli.ChangeUser(user.Name)
 	}
 }
+
+func EnsureTSB(tsbOC *exutil.CLI) (osbclient.Client, *exec.Cmd) {
+	exists := true
+	if _, err := tsbOC.AdminKubeClient().Extensions().DaemonSets(tsbNS).Get("apiserver", metav1.GetOptions{}); err != nil {
+		if !kerrors.IsNotFound(err) {
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
+		exists = false
+	}
+
+	if !exists {
+		e2e.Logf("Installing TSB onto the cluster for testing")
+		_, _, err := tsbOC.AsAdmin().WithoutNamespace().Run("create", "namespace").Args(tsbNS).Outputs()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		configPath := exutil.FixturePath("..", "..", "examples", "templateservicebroker", "templateservicebroker-template.yaml")
+		stdout, _, err := tsbOC.WithoutNamespace().Run("process").Args("-f", configPath).Outputs()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = tsbOC.WithoutNamespace().AsAdmin().Run("create").Args("-f", "-").InputString(stdout).Execute()
+		// this is weird, but we have to ignore this error in case some of this already exists (race between tests and the like)
+		//o.Expect(err).NotTo(o.HaveOccurred())
+		err = WaitForDaemonSetStatus(tsbOC.AdminKubeClient(), &extensions.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: "apiserver", Namespace: tsbNS}})
+		o.Expect(err).NotTo(o.HaveOccurred())
+	}
+
+	// we're trying to test the TSB, not the service.  We're outside all the normal networks.  Run a portforward to a particular
+	// pod and test that
+	pods, err := tsbOC.AdminKubeClient().Core().Pods(tsbNS).List(metav1.ListOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	var pod *kapiv1.Pod
+	for i := range pods.Items {
+		currPod := pods.Items[i]
+		for _, cond := range currPod.Status.Conditions {
+			if cond.Type == kapiv1.PodReady && cond.Status == kapiv1.ConditionTrue {
+				pod = &currPod
+				break
+			}
+		}
+	}
+	port, err := intframework.FindFreeLocalPort()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	portForwardCmd, _, _, err := tsbOC.AsAdmin().WithoutNamespace().Run("port-forward").Args("-n="+tsbNS, pod.Name, fmt.Sprintf("%d:8443", port)).Background()
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	svc, err := tsbOC.AdminKubeClient().Core().Services(tsbNS).Get("apiserver", metav1.GetOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	tsbclient := osbclient.NewClient(&http.Client{Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}}, fmt.Sprintf("https://localhost:%d%s", port, templateapi.ServiceBrokerRoot))
+
+	// wait to get back healthy from the tsb
+	healthResponse := ""
+	err = wait.Poll(e2e.Poll, 2*time.Minute, func() (bool, error) {
+		resp, err := tsbclient.Client().Get("https://" + svc.Spec.ClusterIP + "/healthz")
```

Comment:
do daemonsets have readiness checks?  this seems like it should be a readiness check in the template.